core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchType.java

```diff
@@ -64,9 +65,13 @@ public static MatchType getMatchType(String matchType, Object conditionValue) {
                 }
                 break;
             default:
+                logger.error("Audience condition \"{}\" uses an unknown match type: {}", audienceCondition, matchType);
                 return new MatchType(matchType, new NullMatch());
         }
 
+        logger.error("Audience condition \"{}\" condition value data type is inapplicable for match type \"{}\"",
```

Comment:
isn't it unreachable code?

Reply:
I will recommend to place these logs after this line in UserAttribute. 
Match matchType = MatchType.getMatchType(match, value, this.toString()).getMatcher();
I can't see any difference in line 69 and 75, so we should use both logs there, we may avoid args to add in this method.